e2e/transfer_test.go

```diff
@@ -208,6 +241,159 @@ func (s *TransferTestSuite) TestMsgTransfer_Timeout_Nonincentivized() {
+
+	t.Run("ensure transfer sending is enabled", func(t *testing.T) {
+		enabled := s.QueryTransferSendEnabledParam(ctx, chainA)
+		s.Require().True(enabled)
+	})
+
+	t.Run("ensure packets can be sent", func(t *testing.T) {
+		transferTxResp, err := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(transferTxResp)
+	})
+
+	t.Run("change send enabled parameter to disabled ", func(t *testing.T) {
+		changes := []paramsproposaltypes.ParamChange{
+			paramsproposaltypes.NewParamChange(transfertypes.StoreKey, string(transfertypes.KeySendEnabled), "false"),
+		}
+
+		proposal := paramsproposaltypes.NewParameterChangeProposal(ibctesting.Title, ibctesting.Description, changes)
+		s.ExecuteGovProposal(ctx, chainA, chainAWallet, proposal)
+	})
+
+	t.Run("ensure transfer params are disabled", func(t *testing.T) {
+		enabled := s.QueryTransferSendEnabledParam(ctx, chainA)
+		s.Require().False(enabled)
+	})
+
+	t.Run("ensure ics20 transfer fails", func(t *testing.T) {
+		transferTxResp, err := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0)
+		s.Require().NoError(err)
+		s.Require().Equal(transfertypes.ErrSendDisabled.ABCICode(), transferTxResp.Code)
+	})
+}
+
+// TestReceiveEnabledParam tests changing ics20 ReceiveEnabled parameter
+func (s *TransferTestSuite) TestReceiveEnabledParam() {
+	t := s.T()
+	ctx := context.TODO()
+
+	relayer, channelA := s.SetupChainsRelayerAndChannel(ctx, transferChannelOptions())
+	chainA, chainB := s.GetChains()
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+
+	var (
+		chainBDenom    = chainB.Config().Denom
+		chainAIBCToken = s.getIBCToken(chainBDenom, channelA.PortID, channelA.ChannelID) // IBC token sent to chainA
+
+		chainAAddress = chainAWallet.Bech32Address(chainA.Config().Bech32Prefix)
+		chainBAddress = chainBWallet.Bech32Address(chainB.Config().Bech32Prefix)
+	)
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("ensure transfer receive is enabled", func(t *testing.T) {
+		enabled := s.QueryTransferReceiveEnabledParam(ctx, chainA)
+		s.Require().True(enabled)
+	})
+
+	t.Run("ensure packets can be received, send from chainB to chainA", func(t *testing.T) {
+		t.Run("send from chainB to chainA", func(t *testing.T) {
+			transferTxResp, err := s.Transfer(ctx, chainB, chainBWallet, channelA.Counterparty.PortID, channelA.Counterparty.ChannelID, testvalues.DefaultTransferAmount(chainBDenom), chainBAddress, chainAAddress, s.GetTimeoutHeight(ctx, chainA), 0)
+			s.Require().NoError(err)
+			s.AssertValidTxResponse(transferTxResp)
```

Comment:
This technically does not assert that the transaction succeeded. When it does, it'll probably be unnecessary to assert that the tokens were escrowed